apiserver/clusterservice/scaleimpl.go

```diff
@@ -292,49 +291,48 @@ func ScaleDown(deleteData bool, clusterName, replicaName, ns string) msgs.ScaleD
 		return response
 	}
 
-	//if this was the last replica then remove the replica service
-	var replicaList *v1.DeploymentList
-	selector := config.LABEL_PG_CLUSTER + "=" + clusterName + "," + config.LABEL_SERVICE_NAME + "=" + clusterName + "-replica"
-	replicaList, err = kubeapi.GetDeployments(apiserver.Clientset, selector, ns)
+	// selector in the format "pg-cluster=<cluster-name>,role=replica"
+	// which will grab any/all replicas
+	selector := fmt.Sprintf("%s=%s,%s=%s", config.LABEL_PG_CLUSTER, clusterName,
+		config.LABEL_PGHA_ROLE, "replica")
+	replicaList, err := kubeapi.GetPods(apiserver.Clientset, selector, ns)
 	if err != nil {
 		response.Status.Code = msgs.Error
 		response.Status.Msg = err.Error()
 		return response
 	}
-	if len(replicaList.Items) == 0 {
-		response.Status.Code = msgs.Error
-		response.Status.Msg = "no replicas found for this cluster"
-		return response
-	}
```

Comment:
We should not remove this on 4.2. This would technically be a breaking API change. It's ok to remove in 4.3

Reply:
I'll go ahead and add this back.